lib/dependencies/HarmonyImportSpecifierDependency.js

```diff
@@ -310,14 +315,71 @@ HarmonyImportSpecifierDependency.Template = class HarmonyImportSpecifierDependen
 		// Skip rendering depending when dependency is conditional
 		if (connection && !connection.isTargetActive(runtime)) return;
 
-		const ids = dep.getIds(moduleGraph);
-		const exportExpr = this._getCodeForIds(dep, source, templateContext, ids);
-		const range = dep.range;
+		const ids = dep.getIds(moduleGraph); // determine minimal set of IDs.
+		const trimmedIds = this._trimIdsToThoseImported(ids, moduleGraph, dep);
+		const exportExpr = this._getCodeForIds(
+			dep,
+			source,
+			templateContext,
+			trimmedIds
+		);
+
+		let [rangeStart, rangeEnd] = dep.range;
+		if (trimmedIds.length !== ids.length) {
+			// The array returned from dep.idRangeStarts is right-aligned with the array returned from dep.getIds.
+			// Meaning, the two arrays may not always have the same number of elements, but the last element of
+			// dep.idRangeStarts corresponds to [the starting range position of] the last element of dep.getIds.
+			// Use this to find the correct range end position based on the number of ids that were trimmed.
+			const idx = dep.idRangeStarts.length + (trimmedIds.length - ids.length);
+			if (idx < 0 || idx >= dep.idRangeStarts.length) {
+				throw new Error(
+					`Missing range starts data for id replacement trimming.`
+				);
+			}
```

Comment:
Can we do behaviour more safe? i.e. do nothing if it is missing, because plugins can use it and we will break it

Reply:
Trimming isn't really optional.  If other code is later added to webpack that doesn't pass in the range starts array, then the functionality of "Works with Closure advanced minification" gets broken.
Is there a way to surface a warning instead?
And can it then be turned into an error for webpack v6?

Reply:
> Is there a way to surface a warning instead?

Because any warnings in production will failed to build, we can't, we can log it, because logs are safe

> And can it then be turned into an error for webpack v6?

Yes, we can, but only for webpack v6, now it can be unsafe due plugins

Reply:
Fixed.  Is there something more to do beyond adding the "TODO webpack 6" comment?